From ccf223b424b296f9e4937906ca8a3c40dd467f35 Mon Sep 17 00:00:00 2001
From: Jonathan Neal <jonathantneal@hotmail.com>
Date: Tue, 15 Sep 2015 19:12:40 -0400
Subject: [PATCH 14/99] Update plugin

Support prefix option
Add document and supports to bubbles
Update documentation
---
 README.md | 17 +++++++++++++++++
 index.js  |  7 +++++--
 2 files changed, 22 insertions(+), 2 deletions(-)

diff --git a/README.md b/README.md
index ac870ab..85f4386 100644
--- a/README.md
+++ b/README.md
@@ -136,6 +136,23 @@ grunt.initConfig({
 });
 ```
 
+
+## Options
+
+#### `bubble`
+
+Type: `Array`  
+Default: `['document', 'media', 'supports']`
+
+Specifies additional at-rules whose contents should be transpiled so that the at-rule comes first. By default, `@media`, `@supports` and `@document` will do this.
+
+#### `prefix`
+
+Type: `String`  
+Default: `null`
+
+Specifies a prefix to be surrounded by dashes before the `@nest` at-rule (e.g. `@-x-nest`).
+
 [ci]: https://travis-ci.org/jonathantneal/postcss-nesting
 [ci-img]: https://travis-ci.org/jonathantneal/postcss-nesting.svg
 [Gulp PostCSS]: https://github.com/postcss/gulp-postcss
diff --git a/index.js b/index.js
index 490226a..e6ba6fd 100644
--- a/index.js
+++ b/index.js
@@ -1,5 +1,6 @@
-var bubble  = ['media'];
+var bubble  = ['document', 'media', 'supports'];
 var postcss = require('postcss');
+var prefix  = '';
 
 function transpileSelectors(fromRule, toRule) {
 	var selectors = [];
@@ -56,6 +57,7 @@ function transpileAtRule(fromRule, toRule, atRule) {
 function transpileRule(rule) {
 	var nodes = rule.nodes;
 	var index = -1;
+	var name  = prefix ? '-' + prefix + '-nest' : 'nest';
 	var child;
 
 	// for each node
@@ -65,7 +67,7 @@ function transpileRule(rule) {
 			var newRule = postcss.rule();
 
 			// if atrule is nest
-			if (child.name === 'nest') {
+			if (child.name === name) {
 				transpileNestRule(rule, newRule, child);
 
 				transpileRule(newRule);
@@ -82,6 +84,7 @@ function transpileRule(rule) {
 
 module.exports = postcss.plugin('postcss-nested', function (opts) {
 	if (opts && opts.bubble) bubble = bubble.concat(opts.bubble);
+	if (opts && opts.prefix) prefix = opts.prefix;
 
 	return function (css) {
 		var nodes = css.nodes;
-- 
2.33.1

